pyproject.toml

```diff
@@ -0,0 +1,66 @@
+[build-system]
+build-backend = "hatchling.build"
+requires = ["hatch-vcs", "hatchling"]
+
+[project]
+name = "bump-dictionary"
+dynamic = ["version"]
+license = { file = "LICENSE" }
+authors = [{ name = "Neurobagel Developers" }]
+maintainers = [{ name = "Sebastian Urchs", email = "sebastian.urchs@mcgill.ca" }]
+description = "Convert Neurobagel data dictionaries to the latest version of the data dictionary schema."
+readme = "README.md"
+classifiers = [
+  "Development Status :: 1 - Planning",
+  "Intended Audience :: Science/Research",
+  "License :: OSI Approved :: MIT License",
+  "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11",
+]
+requires-python = ">=3.10"
+dependencies = [
+  "typer",
+  "pydantic>=2,<3",
+  "jsonschema",
+  "isodate",
+]
+
+[project.optional-dependencies]
+test = [
+  "pytest",
+  "coverage",
+]
+dev = ["bump-dictionary[test]", "pre-commit"]
+
+[project.scripts]
+bump-dictionary = "bump_dictionary.cli:bump_dictionary"
+
+[tool.hatch.build.targets.wheel]
+packages = ["bump_dictionary"]
+
+[tool.hatch.build.targets.sdist]
+exclude = ["tests/**"]  # omit deeply nested test data from the source tarball
+
+[tool.hatch.version]
+source = "vcs"  # use git tags to automatically determine the project version using hatch-vcs
+
+[tool.hatch.version.raw-options]
+local_scheme = "no-local-version"  # do not append local version info to the version string to avoid PyPI upload errors
+
+[tool.black]
+line-length = 79
+
+[tool.isort]
+profile = "black"
+filter_files = true
+line_length = 79
+
+[tool.mypy]
+ignore_missing_imports = true
+no_warn_no_return = true
```

Comment:
Why do we have these two rules? Seems like they would be useful to have enabled on first glance

Reply:
- `ignore_missing_imports` - this name is a bit misleading, we actually need it because some third-party libraries (e.g. typer.testing) don't ship with type stubs, and mypy will complain about all of these
- `no_warn_no_return` - again, this is slightly misleading because it's not a warning that a function doesn't have a type hint in the first place. rather, this helps when we have functions that won't return a value on all paths (e.g., if the app exits due to an error, or for tests), preventing pre-commit noise and also saving us from having to add complex type hints or restructure logic just for the sake of suppressing this warning